.github/workflows/schools_updater.yml

```diff
@@ -0,0 +1,77 @@
+name: Fetch Paginated JSON and Update CSV
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs once a day at midnight (UTC)
+  workflow_dispatch:    # Allows for manual runs as well
+  push:
+    branches:
+      - aa_schools_csv_updater
+
+jobs:
+  update_csv:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install jq and csvkit
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y jq
+          pip install csvkit
+
+      - name: Fetch paginated JSON data from API
+        env:
+          API_URL: ${{ secrets.API_URL }}
+          API_TOKEN: ${{ secrets.API_TOKEN }}
+        run: |
+          # Initialize page and limit parameters
+          PAGE=1
+          LIMIT=250
+          AGGREGATE_FILE=data.json
+          echo "[]" > $AGGREGATE_FILE  # Initialize an empty JSON array
+
+          while true; do
+            # Fetch a page of data from the API
+            RESPONSE=$(curl -s -H "Authorization: Bearer $API_TOKEN" "$API_URL?page=$PAGE&limit=$LIMIT")
+
+            # Parse response and check if it's empty
+            PAGE_DATA=$(echo "$RESPONSE" | jq '.')
+            if [ "$(echo "$PAGE_DATA" | jq 'length')" -eq 0 ]; then
+              break
+            fi
+
+            # Append the page data to the aggregate JSON array
+            jq -s '.[0] + .[1]' $AGGREGATE_FILE <(echo "$PAGE_DATA") > tmp.json && mv tmp.json $AGGREGATE_FILE
+
+            # Increment page number for the next request
+            PAGE=$((PAGE + 1))
+          done
+
+      - name: Convert JSON to CSV
+        run: |
+          # Convert JSON to CSV and save as school.csv
+          in2csv data.json > schools.csv
```

Comment:
We only actually care about the name column, particularly given the size of the CSV.